strawberryfields/backends/fockbackend/ops.py

```diff
@@ -417,20 +406,21 @@ def entry(n):
 
 
 @functools.lru_cache()
-def displacedSqueezed(alpha, r, phi, trunc):
+def displacedSqueezed(r_d, phi_d, r_s, phi_s, trunc):
     r"""
     The displaced squeezed state :math:`\ket{\alpha,\zeta} = D(\alpha)S(r\exp{(i\phi)})\ket{0}`.
     """
-    if r == 0:
-        return coherentState(alpha, trunc)
+    if np.allclose(r_s, 0.0):
```

Comment:
Just curious, were there cases of small numeric errors such that checking with `np.allclose` was introduced?

Reply:
Normally it's not a good idea to compare a float to zero directly. Using np.allclose, or np.isclose it's the recommended way.